icaruscode/Timing/PMTTimingCorrectionsProvider.cxx

```diff
@@ -10,114 +10,92 @@
-            << "Calibration database access failed. URL (" << url
-            << "). HTTP error status: " << getHTTPstatus(dataset) 
-            << ". HTTP error message: " << getHTTPmessage(dataset); 
-    }
+uint64_t icarusDB::PMTTimingCorrectionsProvider::RunToDatabaseTimestamp( uint32_t run ) {
 
-    return error;
+   // Run number to timestamp used in the db
+   // DBFolder.h only takes 19 digit (= timestamp in nano second),
+   // but ICARUS tables are currently using run numbers
+   // Step 1) Add 1000000000 to the run number; e.g., run XXXXX -> 10000XXXXX
+   // Step 2) Multiply 1000000000
+   uint64_t runNum = uint64_t(run);
+   uint64_t timestamp = runNum+1000000000;
+   timestamp *= 1000000000;
 
+   if( fVerbose ) mf::LogInfo(fLogCategory) << "Run " << runNum << " corrections from DB timestamp " << timestamp << std::endl;
+   
+   return timestamp;
 }
 
-
-// -----------------------------------------------------------------------------
+// -------------------------------------------------------------------------------
 
 /// Function to look up the calibration database at the table holding the pmt hardware cables corrections
 void icarusDB::PMTTimingCorrectionsProvider::ReadPMTCablesCorrections( uint32_t run ) { 
 
     // pmt_cables_delay: delays of the cables relative to trigger 
     // and reset distribution
-    const std::string name("pmt_cables_delays_data");
-    Dataset dataset;
-    int error = ConnectToDataset( name, run, dataset );
-
-    if (error) throw(std::exception());
-
-    for( int row=4; row < getNtuples(dataset); row++ ) {
-
-        Tuple tuple = getTuple(dataset, row);
-
-        if( tuple != NULL ) {
-
-            unsigned int channel_id = getLongValue( tuple, 0, &error );
-            if( error ) throw std::runtime_error( "Encountered error while trying to access 'channel_id' on table " + name );
-            
-            // PPS reset correction
-            double reset_distribution_delay = getDoubleValue( tuple, 11, &error );
-            if( error ) throw std::runtime_error( "Encountered error '" + std::to_string(error) + "' while trying to access 'reset_distribution_delay' on table " + name );
-
-            // Trigger cable delay
-            double trigger_reference_delay = getDoubleValue( tuple, 10, &error );
-            if( error ) throw std::runtime_error( "Encountered error while trying to access 'trigger_reference_delay' on table " + name );
-
-            // Phase correction
-            double phase_correction = getDoubleValue( tuple, 13, &error );
-            if( error ) throw std::runtime_error( "Encountered error while trying to access 'phase_correction' on table " + name );
+    const std::string dbname("pmt_cables_delays_data");
+    lariov::DBFolder db(dbname, "", "", fTag, true, false);
+
+    bool ret = db.UpdateData( RunToDatabaseTimestamp(run) ); // select table based on run number   
+    if( !ret ) throw std::runtime_error( "Unable to find or open " + dbname + ".db");
```

Comment:
Have you tested this running on multiple files?
It looks like `DBFolder::UpdateData()` returns `false` if the cache was not updated, that should happen also when the cache is already current. When a new file is opened, _art_ believes a new run is being processed, so this method may end up being called also on file boundaries, not only on run boundaries.
My suggestion is to ignore the return value, or rather:
```suggestion
    mf::LogDebug(fLogCategory) << "PMT corrections" << (ret? "": " not") << " updated for run " << run;
```

Reply:
I repeated the test, running of multiple files (same run). The corrections are read again when the next file is opened, and no exception is thrown. I originally added this check as a way to catch missing `.db` files, but I noticed that if a file is missing `DBFolder::UpdateData()` trips internally on its own (producing an exception) before being able to return `false`. So I believe your suggestion to just log it is the best course of actions as I have no intention of changing `DBFolder`.